compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/writer/ConstantPool.java

```diff
@@ -41,16 +43,18 @@
     // Size to be written to tag a null value
     private static final int NULL_VALUE_FIELD_SIZE_TAG = -1;
 
+    private final Map<CPEntry, Integer> cpEntriesMap = new HashMap<>();
     private final List<CPEntry> cpEntries = new ArrayList<>();
```

Comment:
Shall we remove this and maintain the only cpEntriesMap?

Reply:
As writeToStream method modify this collection while iterating over it it's difficult to just use a map/set here. One thing we can do is encapsulate this Map and List in a separate class.